src/hotspot/share/nmt/vmatree.hpp

```diff
@@ -167,14 +171,20 @@ class VMATree {
     }
   };
 
-  SummaryDiff register_mapping(position A, position B, StateType state, const RegionData& metadata);
+ private:
+  SummaryDiff register_mapping(position A, position B, StateType state, const RegionData& metadata, bool copy_flag = false);
 
+ public:
   SummaryDiff reserve_mapping(position from, position sz, const RegionData& metadata) {
-    return register_mapping(from, from + sz, StateType::Reserved, metadata);
+    return register_mapping(from, from + sz, StateType::Reserved, metadata, false);
+  }
+
+  SummaryDiff commit_mapping(position from, position sz, const RegionData& metadata, bool copy_flag = false) {
+    return register_mapping(from, from + sz, StateType::Committed, metadata, copy_flag);
   }
```

Comment:
Why do we need this API? It's only used in the test so far, so do we really need it?

I am not too crazy about `register_mapping` basically being a wrapper around `register_mapping(.., true)`

What if someone calls `commit_mapping(.., true)` ? Doesn't that turn into `register_mapping(..., StateType::Committed, metadata, true)` which is the same as `uncommit_mapping()` ? It doesn't make sense?

Reply:
`StateType::Reserved` is used instead.